### ISSUE TITLE
webhook notification

### ISSUE THREAD

**Issue by mazhigali** [jasonmunro/cypht#178]
Can you add webhook notification while new email comes? 
I would like to use telegram notifications.
Thank you for excellent application and great job

**Comment by jasonmunro** [comment]
Thank you for your kind words!
Interesting idea for a module set. Just to clarify, when new messages are detected we would do some sort of authenticated POST to a webhook for a system like Telegram? I presume this is the API we would want to leverage for that: https://core.telegram.org/bots ?

**Comment by mazhigali** [comment]
Yes you are right. I'm using this script https://github.com/rakshazi/telegram-notify. But I think within the module sypth you can come up with a more functional solution. For example, in the message to send the body of the letter and header, And also be able to remove the email from the telegrams like this @GmailBot

**Comment by marclaporte** [comment]
For the record:
https://en.wikipedia.org/wiki/Webhook

FYI, we just added Webhooks to Tiki Wiki CMS Groupware:
https://gitlab.com/tikiwiki/tiki/-/merge_requests/1649/

**Comment by marclaporte** [comment]
Extending this feature request: I would like notifications for emails that fit a specific criteria. So we could add this to https://github.com/cypht-org/cypht/tree/master/modules/saved_searches

**Comment by marclaporte** [comment]
@mazhigali We will soon have something ready for testing: https://github.com/cypht-org/cypht/pull/1056

**Comment by airbornelamb** [comment]
@marclaporte this is very exciting. Will this work only for telegram or generic webhooks as well?

Generic webhooks would allow Zapier/Make.com usage so that people can shoot the notifications off to other places as they desire. Opens up some powerful use cases.

**Comment by marclaporte** [comment]
@airbornelamb Thank you for the kind words.

I agree. Can you describe a realistic use case so @Shadow243 can handle it?

**Comment by airbornelamb** [comment]
For sure, the biggest use case in front of me would be to send the notification off to Make.com using their generic webhook catch in real-time or near real-time from the message being received. Let me describe in more JTBD...

I am an email power user that has Google workspace accounts at a few companies. Many of them are lower traffic but still demand an urgent response (Within 15 mins ideally). These accounts have been configured in Cypht via Oauth. As a power user, who is frequently on my mobile device, I would like Cypht to send a webhook payload upon receiving an email with the following data: sender, recipient, email subject, email body. I will then utilize Make/Zapier to receive this payload and pass to an alerting service, such as Pushover to alert my phone. This allows me to read incoming emails on mobile without signing into all the accounts or setting up any apps. I can then go to the desktop to respond to important ones.

My most important feature would be real-time alerting of incoming mail.

**Comment by Shadow243** [comment]
> For sure, the biggest use case in front of me would be to send the notification off to Make.com using their generic webhook catch in real-time or near real-time from the message being received. Let me describe in more JTBD...
> 
> 
> 
> I am an email power user that has Google workspace accounts at a few companies. Many of them are lower traffic but still demand an urgent response (Within 15 mins ideally). These accounts have been configured in Cypht via Oauth. As a power user, who is frequently on my mobile device, I would like Cypht to send a webhook payload upon receiving an email with the following data: sender, recipient, email subject, email body. I will then utilize Make/Zapier to receive this payload and pass to an alerting service, such as Pushover to alert my phone. This allows me to read incoming emails on mobile without signing into all the accounts or setting up any apps. I can then go to the desktop to respond to important ones.
> 
> 
> 
> My most important feature would be real-time alerting of incoming mail.

Interesting, I'm gonna try it https://www.make.com/en/api-documentation

**Comment by marclaporte** [comment]
@airbornelamb Do your servers support https://en.wikipedia.org/wiki/IMAP_IDLE ?

**Comment by airbornelamb** [comment]
> @airbornelamb Do your servers support https://en.wikipedia.org/wiki/IMAP_IDLE ?

I'm using Google Workspace over Oauth (following the Cypht guide).. so not sure if it uses IDLE

**Comment by marclaporte** [comment]
> not sure if it uses IDLE

Ok, we will add an easy way to check: https://github.com/cypht-org/cypht/pull/1148

**Comment by airbornelamb** [comment]
Can beta test whenever you get to that point.

**Comment by marclaporte** [comment]
> Can beta test whenever you get to that point.

https://github.com/cypht-org/cypht/pull/1148 is available since Cypht 2.2.0 (we are now at 2.4.0)

**Comment by muava12** [comment]
is this implemented?
